Dockerfile

```diff
@@ -25,15 +25,41 @@ ENV PATH=${PATH}:/conda/bin
 SHELL ["/bin/bash", "-c"]
 
 # Build cuDF conda env
-ADD conda /cudf/conda
-RUN conda env create --name cudf --file /cudf/conda/environments/dev_py35.yml
+ARG PYTHON_VERSION
+ENV PYTHON_VERSION=$PYTHON_VERSION
+ARG NUMBA_VERSION
+ENV NUMBA_VERSION=$NUMBA_VERSION
+ARG NUMPY_VERSION
+ENV NUMPY_VERSION=$NUMPY_VERSION
+ARG PANDAS_VERSION
+ENV PANDAS_VERSION=$PANDAS_VERSION
+ARG PYARROW_VERSION
+ENV PYARROW_VERSION=$PYARROW_VERSION
+ARG CYTHON_VERSION
+ENV CYTHON_VERSION=$CYTHON_VERSION
+ARG CMAKE_VERSION
+ENV CMAKE_VERSION=$CMAKE_VERSION
+ARG CUDF_REPO=https://github.com/rapidsai/cudf
+ENV CUDF_REPO=$CUDF_REPO
+ARG CUDF_BRANCH=master
+ENV CUDF_BRANCH=$CUDF_BRANCH
 
-# Preserved for users who currently use these build-args
-# Clone cuDF repo
-#ARG CUDF_REPO=https://github.com/rapidsai/cudf
-#ARG CUDF_BRANCH=master
-#RUN git clone --recurse-submodules -b ${CUDF_BRANCH} ${CUDF_REPO} /cudf
-ADD cpp /cudf/cpp
+# Always adds the Dockerfile and conditionally adds the local files, at the end we remove the
+# Dockerfiles from each folder and then check if the local files existed, if they did use them, if
+# not clean up folder and clone based on build-args
+ADD Dockerfile cpp* /cudf/cpp/
+ADD Dockerfile python* /cudf/python/
+ADD Dockerfile docs* /cudf/docs/
+ADD Dockerfile .git* /cudf/.git/
+ADD Dockerfile docker* /cudf/docker/
+ADD Dockerfile conda* /cudf/conda/
```

Comment:
Instead of doing all of this craziness, we're going to update this to add the whole build context, and then try to run the `package_versions.sh` script, if it fails then empty the `/cudf` folder, git clone into it, then run the `package_versions.sh` script again.